tests/realtikvtest/pessimistictest/pessimistic_test.go

```diff
@@ -3367,4 +3367,40 @@ func TestPointLockNonExistentKeyWithAggressiveLockingUnderRC(t *testing.T) {
 	tk.MustExec("begin pessimistic")
 	tk.MustExec("select * from t where a = 1 for update")
 	tk.MustExec("commit")
+
+	lockedWithConflictCounter := 0
+	tk2 := testkit.NewTestKit(t, store)
+	tk2.MustExec("use test")
+
+	// Test key exist + write conflict, and locking with conflict takes effect.
+	tk.MustExec("begin pessimistic")
+	tk2.MustExec("begin pessimistic")
+	tk2.MustExec("insert into t values (1, 2)")
+	require.NoError(t, failpoint.EnableWith("github.com/pingcap/tidb/store/driver/txn/lockedWithConflictOccurs", "return", func() error {
+		lockedWithConflictCounter++
+		return nil
+	}))
+	defer func() {
+		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/store/driver/txn/lockedWithConflictOccurs"))
+	}()
+	ch := mustQueryAsync(tk, "select * from t where a = 1 for update")
+	mustTimeout(t, ch, time.Millisecond*50)
```

Comment:
`50ms` may be a little bit short in some extreme cases, how about `200ms`.

Reply:
Since it always blocks, I'm kinda afraid it may take too long time to finish te test. I've updated it to 100ms which is also used in many other test cases.